File: typescript/7kyu/Trimming_a_string.ts
// export const trim = (str: string, size: number): string => {
//   if (str.length <= size) return str;
//   if (size <= 3) return `${str.slice(0, size)}...`;
//   else return `${str.slice(0, size - 3)}...`;
// };

export const trim = (str: string, size: number): string =>
  str.length <= size
    ? str
    : size <= 3
    ? `${str.slice(0, size)}...`
    : `${str.slice(0, size - 3)}...`;

File: rust/src/7kyu/Trimming_a_string.rs
fn trim(phrase: &str, length: usize) -> String {
    let mut res = phrase.to_string();
    if res.len() <= length {
        return res;
    } else if res.len() > length && length <= 3 {
        res.truncate(length);
        res.push_str("...");
    } else {
        res.truncate(length - 3);
        res.push_str("...");
    }
    res
}
